tooling/autodispose-rx-error-prone-checker/src/main/java/com/uber/autodispose/rx/error/prone/checker/RxJavaMissingAutoDisposeErrorChecker.java

```diff
@@ -0,0 +1,197 @@
+        public boolean matches(ExpressionTree tree, VisitorState state) {
+          if (!(tree instanceof MethodInvocationTree)) {
+            return false;
+          }
+          MethodInvocationTree invTree = (MethodInvocationTree) tree;
+
+          final MemberSelectTree memberTree = (MemberSelectTree) invTree.getMethodSelect();
+          if (!memberTree.getIdentifier().contentEquals(TO)) {
+            return false;
+          }
+
+          for (MethodMatchers.MethodNameMatcher nameMatcher : TO_CALL_MATCHERS) {
+            if (nameMatcher.matches(invTree, state)) {
+              ExpressionTree arg = invTree.getArguments().get(0);
+              final Type scoper = state.getTypeFromString("com.uber.autodispose.Scoper");
+              return ASTHelpers.isSubtype(ASTHelpers.getType(arg), scoper, state);
+            }
+          }
+
+          for (MethodMatchers.MethodNameMatcher nameMatcher : AS_CALL_MATCHERS) {
+            if (nameMatcher.matches(invTree, state)) {
+              ExpressionTree arg = invTree.getArguments().get(0);
+              final Type scoper = state
+                  .getTypeFromString("com.uber.autodispose.AutoDisposeConverter");
+              return ASTHelpers.isSubtype(ASTHelpers.getType(arg), scoper, state);
+            }
+          }
+          return false;
+        }
+      };
+
+  private static Matcher<MethodInvocationTree> matcher(List<String> classesWithLifecycle) {
+    return new Matcher<MethodInvocationTree>() {
+      @Override
+      public boolean matches(MethodInvocationTree tree, VisitorState state) {
+
+        boolean matchFound = false;
+        try {
+          final MemberSelectTree memberTree = (MemberSelectTree) tree.getMethodSelect();
+          if (!memberTree.getIdentifier().contentEquals(SUBSCRIBE)) {
+            return false;
+          }
+
+          for (MethodMatchers.MethodNameMatcher nameMatcher : SUBSCRIBE_MATCHERS) {
+            if (!nameMatcher.matches(tree, state)) {
+              continue;
+            } else {
+              matchFound = true;
+              break;
+            }
+          }
+          if (!matchFound) {
+            return false;
+          }
+
+          ClassTree enclosingClass =
+              ASTHelpers.findEnclosingNode(state.getPath(), ClassTree.class);
+          Type.ClassType enclosingClassType = ASTHelpers.getType(enclosingClass);
+
+          for (String s : classesWithLifecycle) {
+            Type lifecycleType = state.getTypeFromString(s);
+            if (ASTHelpers.isSubtype(enclosingClassType, lifecycleType, state)) {
+              return !METHOD_NAME_MATCHERS.matches(memberTree.getExpression(), state);
+            }
+          }
+          return false;
+        } catch (ClassCastException e) {
+          return false;
+        }
+      }
+    };
+  }
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (matcher(classesWithLifecycle).matches(tree, state)) {
```

Comment:
The result of `matcher(classesWithLifecycle)` should be cached in an instance field

Reply:
https://github.com/uber/AutoDispose/pull/156/commits/96a8baf8501d68d9ebcea27e4929a95d7f21d274